### PR TITLE
Fix #153: New Rule: No space after #

### LINKED ISSUE #153: Rule suggestion: no spaces around # for maps and records
In Erlang, using spaces before and after the `#` symbol when dealing with maps or records, is valid. Still, it should be discouraged. I believe a rule that would prevent things like this from happening would be a nice addition:

```
Map = #   { this => map}.
Map     #      {}.
```

### PR DESCRIPTION
**Pull request by elbrujohalcon**
I created a new rule for this, but I believe that the same effect can be achieved using `no_space` and something like `rules => [{right, "#"}]`. I'm not 100% sure and this rule seems to have a reasoning on its own, so I would keep it separate. But I would accept it if we decide not to merge this PR, anyway.

### CODE REVIEW

File: doc_rules/elvis_style/no_space_after_pount.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# No Space after `#`
+
+There should be no spaces after the `#` symbol for maps nor records.
+
+In Erlang, using spaces after the `#` symbol when dealing with maps or records, is valid.
+Still, it should be discouraged.
+
+## Examples
+
+```
```
Comment (paulo-ferraz-oliveira) [inline doc_rules/elvis_style/no_space_after_pount.md]:
```suggestion
```erlang
```

File: doc_rules/elvis_style/no_space_after_pount.md [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# No Space after `#`
+
+There should be no spaces after the `#` symbol for maps nor records.
```
Comment (paulo-ferraz-oliveira) [inline doc_rules/elvis_style/no_space_after_pount.md:3]:
```suggestion
There should be no spaces after the `#` symbol for maps or records.
```

File: doc_rules/elvis_style/no_space_after_pount.md [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# No Space after `#`
```
[the remark this answers is not in the document]
Comment (elbrujohalcon) [reply doc_rules/elvis_style/no_space_after_pount.md:1]: https://en.wiktionary.org/wiki/Unsupported_titles/Number_sign

> (English symbol name [number sign](https://en.wiktionary.org/wiki/number_sign#English) or [hash](https://en.wiktionary.org/wiki/hash#English) or [pound sign](https://en.wiktionary.org/wiki/pound_sign#English) or [octothorpe](https://en.wiktionary.org/wiki/octothorpe#English))

[the remark this answers is not in the document]
Comment (elbrujohalcon) [reply doc_rules/elvis_style/no_space_after_pount.md:1]: I can rename this as _octothorpe_ but I think people usually call it _pound_.

@maco you're native speaker, what do you think?

[the remark this answers is not in the document]
Comment (elbrujohalcon) [reply doc_rules/elvis_style/no_space_after_pount.md:1]: Oh, I misunderstood your comment. The file name should be changed. It was a typo.

**Pull request merged**